app/models/container_deployment.rb

```diff
@@ -1,390 +0,0 @@
-class ContainerDeployment < ApplicationRecord
-  include_concern "Automate"
-  belongs_to :deployed_ems, :class_name => 'ManageIQ::Providers::ContainerManager'
-  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem', :inverse_of => :container_deployments
-  belongs_to :automation_task
-  has_many :container_deployment_nodes, :dependent => :destroy
-  has_many :container_volumes, :as => :parent
-  has_many :custom_attributes, :as => :resource, :dependent => :destroy
-  has_many :authentications, :as => :resource, :dependent => :destroy
-  DEPLOYMENT_TYPES = %w(origin openshift-enterprise).freeze
-  serialize :customizations, Hash
-  AUTHENTICATIONS_NAMES = {"AuthenticationAllowAll"      => "AllowAllPasswordIdentityProvider",
-                           "AuthenticationHtpasswd"      => "HTPasswdPasswordIdentityProvider",
-                           "AuthenticationLdap"          => "LDAPPasswordIdentityProvider",
-                           "AuthenticationRequestHeader" => "RequestHeaderIdentityProvider",
-                           "AuthenticationOpenId"        => "OpenIDIdentityProvider",
-                           "AuthenticationGoogle"        => "GoogleIdentityProvider",
-                           "AuthenticationGithub"        => "GitHubIdentityProvider",
-                           "AuthPrivateKey"              => "ssh",
-                           "AuthenticationRhsm"          => "rhsm"}.freeze
```

Comment:
@Fryguy Do you know if these `Authentication*` classes were ever referenced anywhere else?  `AuthPrivateKey` is the exception as it was not added in the original PR https://github.com/ManageIQ/manageiq/pull/7517.

Reply:
I'm looking into this but so far, it looks clean... only the AuthPrivateKey

I grepped locally on this repo from this branch and only see AuthPrivateKey... same for my gems and bundled gems:

```
grep -r -E "AuthenticationAllowAll|AllowAllPasswordIdentityProvider|AuthenticationHtpasswd|HTPasswdPasswordIdentityProvider|AuthenticationLdap|LDAPPasswordIdentityProvider|AuthenticationRequestHeader|RequestHeaderIdentityProvider|AuthenticationOpenId|OpenIDIdentityProvider|AuthenticationGoogle|GoogleIdentityProvider|AuthenticationGithub|GitHubIdentityProvider|AuthPrivateKey|AuthenticationRhsm" /Users/joerafaniello/.gem/ruby/2.4.5/bundler/gems
grep -r -E "AuthenticationAllowAll|AllowAllPasswordIdentityProvider|AuthenticationHtpasswd|HTPasswdPasswordIdentityProvider|AuthenticationLdap|LDAPPasswordIdentityProvider|AuthenticationRequestHeader|RequestHeaderIdentityProvider|AuthenticationOpenId|OpenIDIdentityProvider|AuthenticationGoogle|GoogleIdentityProvider|AuthenticationGithub|GitHubIdentityProvider|AuthPrivateKey|AuthenticationRhsm" /Users/joerafaniello/.gem/ruby/2.4.5/gems
```